### DISCUSSION TITLE
Audit Log Bug in Current Docker Stack

### DISCUSSION THREAD

**Discussion by Michael-Andersen** [dcm4che/dcm4chee-arc-light#4426] [Q&A] [answered]
Hi, I've reported this elsewhere but I'm hoping to bring it to the attention of someone else who uses or maintains this DCM4CHEE/Keycloak/Logstash stack: https://github.com/dcm4che/dcm4chee-arc-light/wiki/Running-on-Docker

The bug is that events from Keycloak show up in the audit logs with the wrong username. I've provided more detail here: https://github.com/dcm4che/dcm4che/issues/1397 

I believe the source of this bug is the dcm4che-audit-keycloak module. I'm just hoping to get some confirmation that this bug exists and will be fixed in a future version. I'm willing to help with a PR even but I want to make sure if I do that someone will review and accept it (if suitable).

**Comment by vrindanayak** [comment, marked as answer]
Thanks for reporting the issue. Your suggested changes have been incorporated together with slight adjustment and also to cater to Keycloak Admin events. Fix will be available in 5.32.0 release.
